packages/react/src/__tests__/Dialog.test.tsx

```diff
@@ -140,12 +140,24 @@ describe('Dialog', () => {
   })
 
   it('Returns focus to returnFocusRef on escape', async () => {
-    const {getByTestId, queryByTestId} = HTMLRender(<Component />)
+    const {getByTestId, queryByTestId, getByText} = HTMLRender(<Component />)
 
     expect(getByTestId('inner')).toBeTruthy()
-    fireEvent.keyDown(document.body, {key: 'Escape'})
-
-    expect(queryByTestId('inner')).toBeNull()
+    const tooltipEl = getByText('Close')
+    // first ESC closes the tooltip
+    const user = userEvent.setup()
+    await user.keyboard('{Escape}')
+    await waitFor(() => {
+      // Wait for the tooltip to close
+      expect(tooltipEl).not.toHaveAttribute(':popover-open')
+    })
+
+    // second ESC closes the dialog and returns focus
```

Comment:
This test is [failing](https://github.com/primer/react/actions/runs/9707822928/job/26793694928?pr=4702#step:7:819) - Not sure how to address this. I am trying to await before pressing the second ESC but it doesn't seem to be working. Any suggestions, thoughts? 🤔